### PR TITLE
Removing all references to napalm-base which is no longer supported.

### LINKED ISSUE #61542: [TECH-DEBT] Note Napalm-base is no longer supported, from Slack with Mircea Ulinic
Note Napalm-base is no longer supported, from Slack with Mircea Ulinic
Network Engineer at DigitalOcean

Need to update this execution module to not rely on it

FYI @s0undt3ch as per request, and Slack conversation with Mircea
mirceaulinic:napalm: 11:33 AM
Hi. Sorry for late reply, haven't been much active over here lately... Napalm-base is no longer supported, napalm 2.5.0 is what you should be looking for. Let me know if you have any questions. :wink:

Daithi 1:25 PM
thanks for the info, building packages to support napalm and napalm-base on Redhat, will drop napalm-base then, doing napalm 2.5.0, almost done. Thanks for getting back to me, created a native minion for Arista, only found out about your work after it was done (based on Salt 2019.2.2), re-invention of the wheel :smile: but that is how life goes.
mirceaulinic:napalm: 1:28 PM
Haha I know the feeling. Cool, looking forward to give it a whirl. Thanks for working on this!

david@david-lap2:~/devcode/dgm-salt-fix/salt/salt$ grep -r napalm-base *
runners/bgp.py:    return (False, 'The napalm-base module could not be imported')
runners/net.py:    from netaddr import IPNetwork  # netaddr is already required by napalm-base
runners/net.py:    return (False, 'The napalm-base module could not be imported')
utils/napalm.py:    # from napalm-base
utils/napalm.py:        #   other than napalm-base.
utils/napalm.py:        # For example, if napalm-base does not satisfy the requirements
utils/napalm.py:        #   we may need to define a custom library on top of napalm-base
utils/napalm.py:            log.error('Falling back to napalm-base')

_Originally posted by @dmurphy18 in https://github.com/saltstack/salt/issues/55222#issuecomment-555195551_

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Removing all references to napalm-base which is no longer supported.

### What issues does this PR fix or reference?
Fixes: #61542 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/utils/napalm.py [hunk 1]
```diff
@@ -42,9 +42,6 @@
     HAS_NAPALM = False
 
 try:
-    # try importing ConnectionClosedException
-    # from napalm-base
-    # this exception has been introduced only in version 0.24.0
     from napalm.base.exceptions import ConnectionClosedException
```
Comment (waynew) [inline salt/utils/napalm.py:45]: **question** shouldn't this whole try/except be removed then?

**Review by waynew — CHANGES_REQUESTED** [review]

**Review by twangboy — DISMISSED** [review]

File: salt/utils/napalm.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
     # pylint: disable=unused-import,no-name-in-module
     import napalm
     import napalm.base as napalm_base
+    from napalm.base.exceptions import ConnectionClosedException
```
Comment (dmurphy18) [inline salt/utils/napalm.py:34]: shouldn' t lines 33 and 34 be getting deleted not added if removing napalm base ?

File: salt/utils/napalm.py [hunk 3]
```diff
@@ -317,14 +305,6 @@ def get_device(opts, salt_obj=None):
     network_device = get_device_opts(opts, salt_obj=salt_obj)
     provider_lib = napalm_base
```
Comment (dmurphy18) [inline salt/utils/napalm.py:306]: why the use of napalm_base here if removing it ?

**Review by dmurphy18 — CHANGES_REQUESTED** [review]

File: salt/utils/napalm.py (hunk 2 above)
Comment (garethgreenaway) [reply salt/utils/napalm.py:34]: @dmurphy18 `napalm-base` was a separate package/library that we were importing.  This one is part of the main napalm library.

**Review by dmurphy18 — DISMISSED** [review]

**Review by waynew — DISMISSED** [review]

**Review by twangboy — APPROVED** [review]

**Pull request merged**